packages/pancake-svg/src/svg.js

```diff
@@ -0,0 +1,122 @@
+		}));
+	});
+};
+
+
+export const CompileAllSvgs = ( compiledAll, SettingsSVG, pkgPath, modulePackage ) => {
+	return new Promise( ( resolve, reject ) => {
+		Promise.all( compiledAll )
+			.catch( error => {
+				Log.error(`SVG: Compiling SVG ran into an error: ${ error }`);
+			})
+			.then( ( svgModulePath ) => {
+
+			//get all svgs
+			const svgs = Fs
+				.readdirSync( `${svgModulePath}` )                              //read all files in the svg folder
+				.filter( name => !name.startsWith('.') )                        //let’s hide hidden files
+				.map( name => Path.normalize(`${ svgModulePath }/${ name }`) ); //making them absolute paths
+
+			//convert svg to png
+			if( SettingsSVG.pngs !== false ) {
+				Log.verbose(`SVG: Converting svgs to PNGs`);
+
+				const location = Path.normalize(`${ pkgPath }/${ SettingsSVG.pngs }/`);
+
+				SvgToPng( svgs, location, compiledAll );
+
+			}
+
+			//create svg sprite
+			if( SettingsSVG.name !== false ) {
+
+				const location = Path.normalize(`${ pkgPath }/${ SettingsSVG.location }/`);
+
+				const spriter = new SVGSpriter({
+					dest: location,
+					log: null, // Logging verbosity (default: no logging),
+					shape: { // SVG shape related options
+						id: { // SVG shape ID related options
+							separator: '--', // Separator for directory name traversal
+							generator: function (name) { return name.split('.')[0]; }, // SVG shape ID generator callback
+							pseudo: '~' // File name separator for shape states (e.g. ':hover')
+						},
+						transform: ['svgo'], // List of transformations / optimizations
+						meta: null, // Path to YAML file with meta / accessibility data
+						align: null, // Path to YAML file with extended alignment data
+						dest: 'opt' // Output directory for optimized intermediate SVG shapes
+					},
+					svg: { // General options for created SVG files
+						xmlDeclaration: true, // Add XML declaration to SVG sprite
+						doctypeDeclaration: true, // Add DOCTYPE declaration to SVG sprite
+						namespaceIDs: true, // Add namespace token to all IDs in SVG shapes
+						namespaceClassnames: true, // Add namespace token to all CSS class names in SVG shapes
+						dimensionAttributes: true // Width and height attributes on the sprite
+					},
+					variables: {}, // Custom Mustache templating variables and functions
+					mode: {
+						symbol: {
+							sprite: SettingsSVG.name,
+						},
+						shapes: false
+					}
+				});
+				Log.verbose(`SVG: Adding sprites to the sprite machine`);
+				svgs.map( svg => {
+					spriter.add( svg, null, Fs.readFileSync( svg, { encoding: 'utf-8' } ) );
+				});
+
+				Log.verbose(`SVG: Compile sprites into spritesheet`);
+				spriter.compile( ( error, result ) => {
+					if( error ) {
+						Log.error( error );
+						return reject( error );
+					}
+
+					for (let mode in result) {
+						for (let resource in result[mode]) {
+							fs.writeFileSync(result[mode][resource].path, result[mode][resource].contents);
```

Comment:
That's a lot of i/o that might benefit from an async call. Use `fs.writeFile`?

Reply:
Also you imported `Fs` in upper case and use it here in lower... should error out

Reply:
Will address.